util/unifyfs/src/unifyfs-rm.c

```diff
@@ -244,6 +248,98 @@ static int wait_server_initialization(unifyfs_resource_t* resource,
     return ret;
 }
 
+/**
+ * @brief wait until data stage operation finishes
+ *
+ * @param resource
+ * @param args
+ *
+ * @return
+ */
+
+enum {
+    UNIFYFS_STAGE_IN = 0,
+    UNIFYFS_STAGE_OUT = 1,
+};
+
+static inline unsigned int estimate_timeout(const char* manifest_file, int op)
```

Comment:
Should we allow the user to specify a good timeout for the file stage?

Reply:
I have actually no idea on this. It's relatively easy for stage-in, because all files already exist. But for stage-out, maybe even some users won't have clear idea about how long it may take.